src/com/oltpbenchmark/api/BenchmarkModule.java

```diff
@@ -1,134 +1,108 @@
-import com.oltpbenchmark.api.Loader.LoaderThread;
 import com.oltpbenchmark.catalog.Catalog;
-import com.oltpbenchmark.catalog.Table;
 import com.oltpbenchmark.types.DatabaseType;
 import com.oltpbenchmark.util.ClassUtil;
+import com.oltpbenchmark.util.SQLUtil;
 import com.oltpbenchmark.util.ScriptRunner;
 import com.oltpbenchmark.util.ThreadUtil;
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.*;
 
 /**
  * Base class for all benchmark implementations
  */
 public abstract class BenchmarkModule {
-    private static final Logger LOG = Logger.getLogger(BenchmarkModule.class);
-
-    /**
-     * Each benchmark must put their all of the DBMS-specific DDLs
-     * in this directory.
-     */
-    public static final String DDLS_DIR = "ddls";
+    private static final Logger LOG = LoggerFactory.getLogger(BenchmarkModule.class);
 
 
-    /**
-     * Each dialect xml file  must put their all of the DBMS-specific DIALECTs
-     * in this directory.
-     */
-    public static final String DIALECTS_DIR = "dialects";
-
-    /**
-     * The identifier for this benchmark
-     */
-    protected final String benchmarkName;
-
     /**
      * The workload configuration for this benchmark invocation
      */
     protected final WorkloadConfiguration workConf;
 
     /**
-     * These are the variations of the Procedure's Statment SQL
+     * These are the variations of the Procedure's Statement SQL
      */
     protected final StatementDialects dialects;
 
-    /**
-     * Database Catalog
-     */
-    protected final Catalog catalog;
-
     /**
      * Supplemental Procedures
      */
-    private final Set<Class<? extends Procedure>> supplementalProcedures = new HashSet<Class<? extends Procedure>>();
+    private final Set<Class<? extends Procedure>> supplementalProcedures = new HashSet<>();
 
     /**
      * A single Random object that should be re-used by all a benchmark's components
      */
     private final Random rng = new Random();
 
-    public BenchmarkModule(String benchmarkName, WorkloadConfiguration workConf, boolean withCatalog) {
-        assert (workConf != null) : "The WorkloadConfiguration instance is null.";
+    private final HikariDataSource dataSource;
+
+    private Catalog catalog = null;
+
+    public BenchmarkModule(WorkloadConfiguration workConf) {
 
-        this.benchmarkName = benchmarkName;
         this.workConf = workConf;
-        this.catalog = (withCatalog ? new Catalog(this) : null);
-        File xmlFile = this.getSQLDialect();
-        this.dialects = new StatementDialects(this.workConf.getDBType(), xmlFile);
+        this.dialects = new StatementDialects(workConf);
+
+        HikariConfig config = new HikariConfig();
```

Comment:
This brings in Hikari

Reply:
Looks reasonably easy to unhikari if that's an issue (just create connections manually)

Reply:
we're ok with that